R/type-vctr.R

```diff
@@ -56,20 +56,37 @@ new_vctr <- function(.data, ..., class = character()) {
   if (!is_vector(.data)) {
     abort("`.data` must be a vector type.")
   }
-  .data <- validate_attr(.data)
+  nms <- validate_names(.data)
 
-  structure(.data, ..., class = c(class, "vctrs_vctr"))
+  restructure(.data, names = nms, ..., class = c(class, "vctrs_vctr"))
```

Comment:
Why can't we use `vec_set_attributes()` here?

Reply:
So you would no longer call `structure()` when >= 3.6? Maybe that would be fine, actually, since `structure()` is just `attributes<-()` with some extra stuff on top.